app/models/services/locators/pubmed_paper_locator.rb

```diff
@@ -6,35 +6,31 @@ def column
   def find_paper
     return super if super
 
+    paper = Paper.new
+
     pubmed = Pubmed.new
     result = pubmed.get_uid_metadata(self.locator_id)
     if (data = result['result'][self.locator_id])
-      names = data['authors'].map {|a| a['name']}
-
-      existing_authors = Author.where(name: names)
-      existing_names = existing_authors.map(&:name)
-      new_authors = (names - existing_names).map {|a| Author.create name: a}
+      paper = self.import_data_to_paper(paper,data,'pubmed')
+    end
 
-      doi = data['elocationid']
-      if doi.blank?
-        doi = data['articleids'].find {|id| id['idtype'] == 'doi' }['value']
+    if paper.doi
+      crossref = Crossref.new
+      if result = crossref.get_doi_metadata(self.locator_id)
+        data = result['message']
+        paper = self.import_data_to_paper(paper,data,'crossref')
       end
+    end
 
-      doi = doi.sub(/^doi: /, "")
+    existing_paper = Paper.where( "title = ? or doi = ? and doi is not null", paper.title, paper.doi )
```

Comment:
`#where` also returns a query - which means semantically that it can return multiple results. Essentially it returns an array that can even be empty.